model_building/xgboost_experiment_configuration.py

```diff
@@ -139,11 +140,16 @@ def print_model(self):
     def initialize_regressor(self):
         """
         Initialize the regressor object for the experiments
+
+        See ExperimentConfiguration class for _disable_model_parallelism
         """
         if not getattr(self, '_hyperparameters', None):
             self._regressor = xgb.XGBRegressor()
         else:
-            self._regressor = xgb.XGBRegressor(min_child_weight=self._hyperparameters['min_child_weight'], gamma=self._hyperparameters['gamma'], n_estimators=self._hyperparameters['n_estimators'], learning_rate=self._hyperparameters['learning_rate'], max_depth=self._hyperparameters['max_depth'], tree_method="hist", objective='reg:squarederror', n_jobs=1)
+            if self._disable_model_parallelism:
+                self._regressor = xgb.XGBRegressor(min_child_weight=self._hyperparameters['min_child_weight'], gamma=self._hyperparameters['gamma'], n_estimators=self._hyperparameters['n_estimators'], learning_rate=self._hyperparameters['learning_rate'], max_depth=self._hyperparameters['max_depth'], tree_method="hist", objective='reg:squarederror', nthread=1, n_jobs=1)
+            else:
+                self._regressor = xgb.XGBRegressor(min_child_weight=self._hyperparameters['min_child_weight'], gamma=self._hyperparameters['gamma'], n_estimators=self._hyperparameters['n_estimators'], learning_rate=self._hyperparameters['learning_rate'], max_depth=self._hyperparameters['max_depth'], tree_method="hist", objective='reg:squarederror', n_jobs=1)
```

Comment:
When trying to run this code, I get the following error coming from here:
```
AttributeError: 'XGBoostExperimentConfiguration' object has no attribute '_disable_model_parallelism'
```
(Obviously this is in debug mode. When in normal mode, the error is simply absorbed by the try-except mechanism and no valid model for XGBoost is built.)

Reply:
One other thing. After reading the [documentation of XGBoost](https://xgboost.readthedocs.io/en/stable/python/python_api.html), I think that you should be changing `n_jobs` instead of `nthread`. What do you think? Does the current code work as intended for you?

Reply:
Looking at the documentation I thought the same but, as you can see, both lines have n_jobs=1: still, the one without nthread=1 tries to use all the core available in the machine it is running on, resulting in problems when parallelism is enabled from the configuration file. For what concerns the error, I will look into it, since it is new to me, possibly I forgot to rename something